packages/card/src/card-link.js

```diff
@@ -0,0 +1,73 @@
+import videojs from 'video.js';
+import { version } from '../package.json';
+import Card from './card.js';
+
+/**
+ * @typedef {object} cardLinkOptions
+ * @property {function(): string} urlHandler function that returns a URL
+ *
+ * @typedef {cardLinkOptions | import('./card.js').CardOptions} CardLinkOptions
+ */
+
+/**
+ * The CardLink functions as a link.
+ *
+ * This component extends the Card class and can be used to navigate to a URL.
+ *
+ * @extends Card
+ */
+class CardLink extends Card {
+  /**
+   * Creates an instance of CardLink.
+   *
+   * @param {import('video.js/dist/types/player').default} player
+   * @param {CardLinkOptions} options
+   */
+  constructor(player, options) {
+    super(player, options);
+
+    /** @type { function(): string } */
+    this.handleUrl = typeof options.urlHandler === 'function' ? options.urlHandler.bind(this) : () => { };
+
+    videojs.dom.setAttributes(this.el(), {
+      href: this.handleUrl(),
+      'aria-labelledby': this.id(),
+      title: this.metadata.title
+    });
+  }
+
+  /**
+   * Builds the CSS class for the component.
+   *
+   * @returns {string} the CSS class name.
+   */
+  buildCSSClass() {
+    return `${super.buildCSSClass()} vjs-card-link`;
+  }
+
+  /**
+   * Creates the main element for the component.
+   *
+   * @returns {HTMLAnchorElement} the main element.
+   */
+  createEl() {
+    return videojs.dom.createEl('a', {
+      className: this.buildCSSClass(),
+      rel: 'noopener noreferrer',
+    });
+  }
```

Comment:
Following the previous comment this can now be implemented as such:

```suggestion
createEl(tag = 'a', props = {}, attributes = {}) {
    return super.createEl(
      tag,
      videojs.obj.merge({
        className: this.buildCSSClass()
      }, props)
      , videojs.obj.merge({
        href: this.options().urlHandler(),
        'aria-labelledby': this.id()
        title: this.options().metadata.title
      }, attributes)
    );
  }
```

Reply:
Your version looks cleaner for sure, I didn't see that you were dependant on the context!